redux-form/lib/reduxForm.d.ts

```diff
@@ -285,65 +268,65 @@ export class SubmissionError<FormData extends DataShape> extends Error {
  * The following are methods or properties that you can access on an instance
  * of your decorated form component (outermost component named "ReduxForm").
  */
-export interface Form<FormData extends DataShape, P, S> extends Component<P, any> {
+export interface FormComponent<FormData extends DataShape, P, S> extends ComponentClass<P> {
```

Comment:
This name change is a BC break, but I see why it's needed.

I'm not sure `FormComponent` is the best name though. The "real" `Form` class is also a component, which makes this ambiguous. Maybe `DecoratedForm` (I'm not sure I like that either)?

Also why make these properties optional?